crates/bevy_math/src/primitives/dim2.rs

```diff
@@ -125,6 +125,94 @@ impl Ellipse {
     }
 }
 
+/// A primitive shape formed by the region between two circles, also known as a ring.
+#[derive(Clone, Copy, Debug, PartialEq)]
+#[cfg_attr(feature = "serialize", derive(serde::Serialize, serde::Deserialize))]
+#[doc(alias = "Ring")]
+pub struct Annulus {
+    /// The inner radius of the annulus
+    pub inner_radius: f32,
+    /// The outer radius of the annulus
+    pub outer_radius: f32,
+}
+impl Primitive2d for Annulus {}
+
+impl Default for Annulus {
+    /// Returns the default [`Annulus`] with radii of `0.5` and `1.0`.
+    fn default() -> Self {
+        Self {
+            inner_radius: 0.5,
+            outer_radius: 1.0,
+        }
+    }
+}
+
+impl Annulus {
+    /// Create a new [`Annulus`] from an inner and outer radius
+    #[inline(always)]
+    pub const fn new(inner_radius: f32, outer_radius: f32) -> Self {
+        Self {
+            inner_radius,
+            outer_radius,
+        }
+    }
+
+    /// Get the diameter of the annulus
+    #[inline(always)]
+    pub fn diameter(&self) -> f32 {
+        2.0 * self.outer_radius
+    }
+
+    /// Get the width of the annulus
+    #[inline(always)]
+    pub fn width(&self) -> f32 {
+        self.outer_radius - self.inner_radius
+    }
```

Comment:
I think this name is misleading. I read this as getting the full width of the ring shape, or the horizontal space it occupies, i.e. the diameter in this case.

Not sure on the ideal name... some ideas:

- `ring_diameter` and `ring_radius`
- `minor_diameter` and `minor_radius` like for a torus (I couldn't find any references with "annulus minor radius" though)
- `thickness`

I think I might prefer the first one since it seems the clearest to me and at least shouldn't be mathematically "incorrect".

Reply:
"thickness" is by far the most clear to me, followed by width. Diameter / radius implies "outer radius" IMO, and "minor_radius" implies an ellipsoid shape.

Reply:
As you might've guessed, I think `width` is the clearest, but `thickness` is fine too: seeing how the consensus is that it should be universally understood (given our generous sample of 3 people :P), I'm gonna update it shortly.